### PR TITLE
Fix asserts in multiplexer plugin.

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

**Comment by ywkaras** [comment]
[approve ci autest]

**Review by randall — DISMISSED** [review]

File: plugins/multiplexer/dispatch.h [hunk 1]
```diff
@@ -31,11 +31,28 @@
 
 #include "ts.h"
 
-#define CHECK(X)                                         \
-  {                                                      \
-    const TSReturnCode r = static_cast<TSReturnCode>(X); \
-    assert(r == TS_SUCCESS);                             \
+#ifdef __OPTIMIZE__
```
Comment (ywkaras) [inline plugins/multiplexer/dispatch.h:34]: I had this backwards (was ifndef).

Comment (ywkaras) [reply plugins/multiplexer/dispatch.h:34]: thanks Brian

**Review by bneradt — APPROVED** [review]
Looks good to me. clang supports the `__OPTIMIZE__` as well as gcc:

https://stackoverflow.com/a/14620320/629530

So we're good on that front.

**Comment by randall** [comment]
[approve ci]

**Comment by ywkaras** [comment]
[approve ci]

File: plugins/multiplexer/dispatch.h [hunk 2]
```diff
@@ -31,12 +31,32 @@
...
+#ifndef NDEBUG
+#define NDEBUG
+#endif
+
+#else
+
+// Check if expression X returns a value that implicitly converts to bool false (such as TS_SUCCESS).
+#define CHECK(X)         \
+  {                      \
+    assert(!TS_SUCCESS); \
```
Comment (SolidWallOfCode) [inline plugins/multiplexer/dispatch.h:54]: Should `static_assert` here.

**Comment by ywkaras** [comment]
@randall I think this is really done now.
https://movieboozer.com/wp-content/uploads/2014/08/lucy-football.jpg

**Review by randall — APPROVED** [review]

**Review by bneradt — APPROVED** [review]

**Comment by zwoop** [comment]
Cherry-picked to v9.1.x branch.

**Pull request merged**